src/core/main.c

```diff
@@ -1275,57 +1275,44 @@ static void bump_file_max_and_nr_open(void) {
 #endif
 
 #if BUMP_PROC_SYS_FS_NR_OPEN
-        int v = INT_MAX;
-
         /* Argh! The kernel enforces maximum and minimum values on the fs.nr_open, but we don't really know
          * what they are. The expression by which the maximum is determined is dependent on the architecture,
          * and is something we don't really want to copy to userspace, as it is dependent on implementation
          * details of the kernel. Since the kernel doesn't expose the maximum value to us, we can only try
          * and hope. Hence, let's start with INT_MAX, and then keep halving the value until we find one that
          * works. Ugly? Yes, absolutely, but kernel APIs are kernel APIs, so what do can we do... 🤯 */
 
-        for (;;) {
-                int k;
-
-                v &= ~(__SIZEOF_POINTER__ - 1); /* Round down to next multiple of the pointer size */
-                if (v < 1024) {
-                        log_warning("Can't bump fs.nr_open, value too small.");
-                        break;
-                }
-
-                k = read_nr_open();
-                if (k < 0) {
-                        log_error_errno(k, "Failed to read fs.nr_open: %m");
-                        break;
-                }
+        /* Max number of open files per process, as per fs/file.c in the kernel source tree, since v2.6:
```

Comment:
maybe move the comment next to the definitions, and clarify that the expression has been stable since kernel v2.6?